pig/src/main/java/org/apache/iceberg/pig/IcebergPigInputFormat.java

```diff
@@ -245,8 +245,14 @@ private boolean advance() throws IOException {
 
     private Object convertPartitionValue(Type type, Object value) {
       if (type.typeId() == Types.BinaryType.get().typeId()) {
-        ByteBuffer buffer = (ByteBuffer) value;
-        return new DataByteArray(buffer.get(new byte[buffer.remaining()]).array());
+        ByteBuffer dupe = ((ByteBuffer) value).duplicate();
+        if (dupe.hasArray()) {
+          return new DataByteArray(dupe.array());
```

Comment:
I don't think that we need to check `hasArray` here. I think the reason why this didn't previously check `hasArray` is that the array passed to `DataByteArray` must start at offset 0 and be valid through the array length, so a copy was needed in almost every case.

It may be simpler to change this to use `ByteBuffers.toByteArray` and pass the result to create `DataByteArray`.

Reply:
Let me know if I used this wrrong but this is a very useful utility function!